src/zkevm_specs/evm/step.py

```diff
@@ -0,0 +1,456 @@
+
+        # Verify callee's new balance is added by value and not overflow
+        bytes_callee_prev_balance = self.decompress(callee_prev_balance, 32, r)
+        bytes_callee_new_balance = self.decompress(callee_new_balance, 32, r)
+        callee_carries = self.allocate_bool(32)
+        assert_addition(bytes_callee_prev_balance, bytes_value, bytes_callee_new_balance, callee_carries)
+        assert callee_carries[31] == 0
+
+    def assert_memory_expansion(
+        self,
+        bytes_cd_offset: Sequence[int],
+        bytes_cd_length: Sequence[int],
+        bytes_rd_offset: Sequence[int],
+        bytes_rd_length: Sequence[int],
+    ) -> Tuple[int, int]:
+        cd_offset = le_to_int(bytes_cd_offset[:5])
+        cd_length = le_to_int(bytes_cd_length)
+        rd_offset = le_to_int(bytes_rd_offset[:5])
+        rd_length = le_to_int(bytes_rd_length)
+
+        next_memory_size = self.allocate(1)[0]
+
+        has_cd_length = not self.is_zero(cd_length)
+        has_rd_length = not self.is_zero(rd_length)
+        bytes_next_memory_size_cd = self.allocate_byte(4)
+        bytes_next_memory_size_rd = self.allocate_byte(4)
+        next_memory_size_cd = has_cd_length * le_to_int(bytes_next_memory_size_cd)
+        next_memory_size_rd = has_rd_length * le_to_int(bytes_next_memory_size_rd)
+
+        # Verify next_memory_size_cd is correct
+        if has_cd_length:
+            assert sum(bytes_cd_offset[5:]) == 0
+        self.fixed_lookup(FixedTableTag.Range32, [32 * next_memory_size_cd - has_cd_length * (cd_offset + cd_length)])
+
+        # Verify next_memory_size_rd is correct
+        if has_rd_length:
+            assert sum(bytes_rd_offset[5:]) == 0
+        self.fixed_lookup(FixedTableTag.Range32, [32 * next_memory_size_rd - has_rd_length * (rd_offset + rd_length)])
+
+        # Verify next_memory_size == \
+        #   max(self.call.memory_size, next_memory_size_cd, next_memory_size_rd)
+        assert next_memory_size in [self.call.memory_size, next_memory_size_cd, next_memory_size_rd]
+        self.bytes_range_lookup(next_memory_size - self.call.memory_size, 4)
+        self.bytes_range_lookup(next_memory_size - next_memory_size_cd, 4)
+        self.bytes_range_lookup(next_memory_size - next_memory_size_rd, 4)
+
+        # Verify memory_gas_cost is correct
+        curr_quad_memory_gas_cost = le_to_int(self.allocate_byte(8))
+        next_quad_memory_gas_cost = le_to_int(self.allocate_byte(8))
+        self.fixed_lookup(FixedTableTag.Range512,
+                          [self.call.memory_size * self.call.memory_size - 512 * curr_quad_memory_gas_cost])
+        self.fixed_lookup(FixedTableTag.Range512,
+                          [next_memory_size * next_memory_size - 512 * next_quad_memory_gas_cost])
+        memory_gas_cost = next_quad_memory_gas_cost - curr_quad_memory_gas_cost + \
+            3 * (next_memory_size - self.call.memory_size)
+
+        return next_memory_size, memory_gas_cost
+
+    def is_jump(self):
```

Comment:
remove this

Reply:
keep is_jump as in assert_transition method requiring it do handle pc property change specially